### PR TITLE
Allow humans to solve captchas.

### PR DESCRIPTION
**Pull request by mcsaucy**
This isn't really needed for Bay County, but St. Johns county (which
also uses Benchmark) uses reCAPTCHA. This should make it easier to
integrate with St Johns county, at least, as well as other Benchmark
sources.

### CODE REVIEW

File: Counties/Florida/Bay County/Scraper/Scraper.py [hunk 1]
```diff
@@ -325,27 +325,36 @@ def search_portal(case_number):
...
-            pass
 
+            # Do search
+            search_button = driver.find_element_by_id('searchButton')
+            search_button.click()
+        else:
+            print(f"Captcha encountered trying to view case ID {case_number}.")
+            print("Please solve the captcha and click the search button to proceed.")
+            WebDriverWait(driver, 60).until(
+                lambda x: case_number in driver.title )
```
Comment (OscarVanL) [inline Counties/Florida/Bay County/Scraper/Scraper.py:350]: If 60 seconds pass here, what happens? I don't see anything to handle the TimeoutException, unless your intended behaviour is for the whole scraper to stop if they don't solve the captcha within 60 seconds?

Comment (mcsaucy) [reply Counties/Florida/Bay County/Scraper/Scraper.py:350]: That's a good question. Presently we just die, and 60 seconds can be too short to solve a reCAPTCHA. We could totally set this to a higher value (maybe an hour?), but it still feels weird to just drop out like that.

What do you think we should do in this case?

Comment (OscarVanL) [reply Counties/Florida/Bay County/Scraper/Scraper.py:350]: In my opinion if the scraper is waiting for the human to solve a captcha, it shouldn't matter how long they take. 
It doesn't make sense for it to die/crash, even if someone left the scraper running for days minimised without ever solving the captcha.

Something [like this](https://stackoverflow.com/a/29383665) with an indefinite wait could do that.

Comment (jameskranz) [reply Counties/Florida/Bay County/Scraper/Scraper.py:350]: How does this mesh with the idea of running a scraper in the cloud?

Comment (mcsaucy) [reply Counties/Florida/Bay County/Scraper/Scraper.py:350]: James: it doesn't mesh. The present directive in CONTRIBUTING.md is:
> For now, while we are still examining the legality of bypassing Captchas and authentication pages, please do not write scrapers that bypass them either programatically or with captcha solving services.

While I agree that's super suboptimal, I don't really have anything else to go off of ATM and this PR allows this scraper to comply with that directive as it stands.

Comment (mcsaucy) [reply Counties/Florida/Bay County/Scraper/Scraper.py]: Updated the webdriver waiting to have limitless patience for slow human captcha solvers.

**Comment by mcsaucy** [comment]
We are now cool with waiting forever for users to solve the captcha.

File: Counties/Florida/Bay County/Scraper/Scraper.py (hunk 1 above)
Comment (OscarVanL) [reply Counties/Florida/Bay County/Scraper/Scraper.py]: If the scraper must run in the cloud, the only way this particular scraper can run is if the `-u`  or  `--solve-captchas` args are given. 

While I did solve the captcha system used by this county with my first PR https://github.com/Police-Data-Accessibility-Project/Police-Data-Accessibility-Project/pull/50, at the current time we're building the scrapers assuming automated solving of captchas isn't allowed (until we have had further guidance that this is allowed). 

So by default, the user will need to solve them by hand. This PR from @mcsaucy adds this functionality.

**Comment by OscarVanL** [comment]
LGTM then!

**Comment by mcsaucy** [comment]
I'll apply this in the Scrapers repo.

**Pull request closed without merging**